openid-4-verifiable-presentations-1_0.md

```diff
@@ -542,6 +542,266 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
 
 If the Verifier responds with any HTTP error response, the Wallet MUST terminate the process.
 
+# Verifiable Presentation Query Language {#vp_query}
+
+The Verifiable Presentation Query Language (VP Query) is a JSON-encoded query
+language that allows the Verifier to request the Wallet to present Verifiable
+Credentials that match the query. The query consists of a set of constraints
+that the requested Verifiable Credentials must satisfy. The Wallet evaluates the
+query against the Verifiable Credentials it holds and returns the Verifiable
+Presentations that match the query.
+
+A valid VP Query is defined as a JSON-encoded object with the following
+properties:
+
+`credentials`:
+: REQUIRED. An object. Each key in the object represents one
+Credential Query as defined in (#credential_query). The key is a string that
+uniquely identifies the Credential Query within the VP Query.
+
+`rules`:
+: OPTIONAL. An object defining rules that specify which combinations of
+credentials MAY be returned by the Wallet, as defined in (#credential_rules). If
+omitted, all credentials listed in `credentials` are requested for presentation.
+
+## Credential Query {#credential_query}
```

Comment:
The credential query structure is defined as a polymorphic structure that contains both elements that always have the same content (purpose), elements that are partly format specific (claims) and elements that are fully format specific (everything else).
Since there is only a single element that is not currently format specific (purpose), we should consider making the request fully profile specific to simplify implementations.

Reply:
assuming by profile specific, you meant format specific? I do not understand how making everything format specific would simplify things... I think the feedback we have been getting is actually in the opposite direction of removing "partly format specific" part and use `path` for mdocs too, instead of `namespaces` and `data_elements` claims. the rest of the structure of defining the framework in a format agnostic way, while allowing format specific content for certain claims (like for the algorithms supported) makes sense to me. path vs namespace/data_elements feels pretty unnatural.